mono/mini/mini-x86.c

```diff
@@ -3223,16 +3244,15 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 
 			if (tailcall_membase_ecx) {
 				x86_jump_membase (code, X86_ECX, ins->inst_offset);
-			} else if (tailcall_membase_not_ecx) {
+			} else if (tailcall_reg || tailcall_membase_not_ecx) {
 				x86_jump_reg (code, X86_ECX);
 			} else {
-				// FIXME alignment
-				mono_add_patch_info (cfg, code - cfg->native_code, MONO_PATCH_INFO_METHOD_JUMP, call->method);
+				// FIXME Patch data instead of code.
+				code = x86_align_and_patch (cfg, code, MONO_PATCH_INFO_METHOD_JUMP, call->method, 1);
 				x86_jump32 (code, 0);
 			}
 
 			ins->flags |= MONO_INST_GC_CALLSITE;
-			cfg->disable_aot = TRUE;
 			break;
```

Comment:
Where is the aot support above ?

Reply:
Oops good point.

Reply:
Arguably it is in x86_align_and_patch but that isn't clear. Let me look again.

Reply:
Ok, I'm convinced, it is x86_align_and_patch.
Perhaps it would be clearer, if emit_call was refactored slightly, to accept a boolean tailcall? Then call and tailcall would share code more clearly?